### PR TITLE
SDIT-4062 OffenderImprisonmentStatus basic rewrite

### PR DESCRIPTION
**Pull request by andymarke**
Add skeleton of main offence update

Added a return class separate from Repository class

Add nested test classes for ImprisonmentStatus and virtual empty for now MainOffenceUpdate 

* TODO locking strategy
* TODO update main offence

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/courtsentencing/ImprisonmentStatusServiceIntTest.kt [hunk 1]
```diff
@@ -54,150 +54,154 @@ class ImprisonmentStatusServiceIntTest {
   }
 
   @Nested
-  inner class ChargeHasNoOutcome {
+  inner class ImprisonmentStatusUpdate {
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/courtsentencing/ImprisonmentStatusServiceIntTest.kt:57]: Nest inside a new parent test

**Review by mikehalmamoj — APPROVED** [review]
